docs/kms.md

```diff
@@ -0,0 +1,67 @@
+# Key Management Services
+
+This document describes how to use a key management service or KMS to store the
+private keys and sign certificates.
+
+Support for multiple KMS are planned, but currently the only supported one is
+Google's Cloud KMS.
+
+## Google's Cloud KMS.
+
+[Cloud KMS](https://cloud.google.com/kms) is the Google's cloud-hosted KMS that
+allows you to store the cryptographic keys, and sign certificates using their
+infrastructure. Cloud KMS supports two different protection levels, SOFTWARE and
+HSM.
+
+To configure Cloud KMS in your CA you need add the `"kms"` property to you
+`ca.json`, and replace the property`"key"` with the Cloud KMS key name of your
+intermediate key:
+
+```json
+{
+    ...
+    "key": "projects/<project-id>/locations/global/keyRings/<ring-id>/cryptoKeys/<key-id>/cryptoKeyVersions/<version-number>",
+    ...
+    "kms": {
+        "type": "cloudkms",
+        "credentialsFile": "path/to/credentials.json"
+    }
+}
+```
+
+In a similar way, for SSH certificate, the SSH keys must be Cloud KMS names:
+
+```json
+{
+    ...
+    "ssh": {
+        "hostKey": "projects/<project-id>/locations/global/keyRings/<ring-id>/cryptoKeys/<key-id>/cryptoKeyVersions/<version-number>",
+        "userKey": "projects/<project-id>/locations/global/keyRings/<ring-id>/cryptoKeys/<key-id>/cryptoKeyVersions/<version-number>"
+    },
+}
+```
+
+Currently [step](https://github.com/smallstep/cli) does not provide an automatic
+way to initialize the public key infrastructure (PKI) using Cloud KMS, but an
+experimental tool named `step-cloudkms-init` is available for this use case. At
+some point this tool will be integrated into `step` and it will be deleted.
+
+The use `step-cloudkms-init` just enable Cloud KMS and run:
```

Comment:
To use `step-cloudkms-init` just enable Cloud KMS in your `step-ca` config and run:

Reply:
Fixed in 8604c31

Reply:
Actually this 334d191, we don't need to enable it in the ca config.